L1Trigger/VertexFinder/src/VertexFinder.cc

```diff
@@ -4,6 +4,118 @@ using namespace std;
 
 namespace l1tVertexFinder {
 
+  double VertexFinder::computeAndSetVertexParametersPFA(RecoVertex<>& vertex) {
+    double pt = 0.;
+    double z0 = -999.;
+    double z0width = 0.;
+    double z0Alt = -999.;
+    double z0widthAlt = 0.;
+
+    bool highPt = false;
+    double highestPt = 0.;
+    unsigned int numHighPtTracks = 0;
+
+    const float sqrt0p5 = std::sqrt(0.5);
+
+    float SumWeight = 0.;
+    float SumWeightedPt = 0.;
+    float SumGaussianAndPtWeight = 0.;
+    float SumZ = 0.;
+    float SumGaussianAndPtWeightAlt = 0.;
+    float SumZAlt = 0.;
+    float z0square = 0.;
+    float z0squareAlt = 0.;
+    float trackPt = 0.;
+    float trackAbsEta = 0.;
+
+    unsigned int itrack = 0;
+
+    for (const L1Track* track : vertex.tracks()) {
+      itrack++;
+      trackPt = track->pt();
+      trackAbsEta = std::fabs(track->eta());
+
+      if (trackPt > settings_->vx_TrackMaxPt()) {
+        highPt = true;
+        numHighPtTracks++;
+        highestPt = (trackPt > highestPt) ? trackPt : highestPt;
+        if (settings_->vx_TrackMaxPtBehavior() == 0)
+          continue;  // ignore this track
+        else if (settings_->vx_TrackMaxPtBehavior() == 1)
+          trackPt = settings_->vx_TrackMaxPt();  // saturate
+      }
+
+      float GaussianWidth = settings_->vx_pfa_etadependentresolution()
+                                ? 0.09867 + 0.0007 * trackAbsEta + 0.0587 * trackAbsEta * trackAbsEta
```

Comment:
What are these numbers? A comment here would be useful to anyone doing maintenance later.